### PR TITLE
Add support to enumerate local changes

### PR DESCRIPTION
**Pull request by khanaffan**
Added `getLocalChanges()` allows the enumeration of all local pending txns or optionally includes in-memory changes. 

* Users can specify a list of root classes to filter the changes polymorphically.
* The function only returns the instance key and changes type and does not return any more details.

itwinjs-core: https://github.com/iTwin/itwinjs-core/tree/affanK/get-local-changes1

### CODE REVIEW

File: iModelCore/iModelPlatform/DgnCore/ChangesetTxns.cpp [hunk 1]
```diff
@@ -466,6 +466,179 @@ void TxnManager::WriteChangesToFile(BeFileNameCR pathname, DdlChangesCR ddlChang
...
+            }
+        }
+    }
+
+    // Group all local txn
+    auto endTxnId = GetCurrentTxnId();
+    auto startTxnId = QueryNextTxnId(TxnId(0));
+    DdlChanges ddlChanges;
+    ChangeGroup dataChangeGroup;
+    for (auto currTxnId = startTxnId; currTxnId < endTxnId; currTxnId = QueryNextTxnId(currTxnId)) {
```
Comment (pmconne) [inline iModelCore/iModelPlatform/DgnCore/ChangesetTxns.cpp:508]: `QueryNextTxnId` will return deleted (undone) txns...you probably want to filter those out.

Comment (pmconne) [reply iModelCore/iModelPlatform/DgnCore/ChangesetTxns.cpp:508]: Nope, I misread. You use `QueryNextTxnId(0)` to get the *first* txn and iterate until >= curr txn. Deleted txns will always be > curr txn.

Comment (khanaffan) [reply iModelCore/iModelPlatform/DgnCore/ChangesetTxns.cpp:508]: Correct, that's what I found out.

**Comment by pmconne** [comment]
Will you submit a corresponding TypeScript PR to provide a beta API?

**Comment by khanaffan** [comment]
> Will you submit a corresponding TypeScript PR to provide a beta API?

done

**Review by pmconne — APPROVED** [review]

**Pull request merged**